### ISSUE TITLE
Modules export minified code

### ISSUE THREAD

**Issue by leanne-tite-jg** [braintree/braintree-web#197]
The UMD modules available in npm export minified code. While this does not stop us using these modules, it throws warnings in our Webpack dev server. Since these are source modules would it  make more sense to export them un-minified and leave the end user to handle minification?

Example of console warning:

```
./~/braintree-web/client.js
Critical dependencies:
1:438-445 This seems to be a pre-built javascript file. Though this is possible, it's not recommended. Try to require the original source to get better results.
@ ./~/braintree-web/client.js 1:438-445
```

**Comment by crookedneighbor** [comment]
The problem is not that it's minified, but that it is pre-built using Browserify. If you use the the debug versions of the files which are not minified (like `require('braintree-web/client.debug')`), you'll get the same error because they are still bundled beforehand.

We're discussing how we can support Webpack without displaying these errors, but it requires us to change how we distribute the files and we want to make sure we don't break anyone who is already using the package. For example, if someone is downloading the module through npm and then copying the files from the node_modules folder directly to their frontend, providing pre-built files would break their integration as soon as they upgraded to a new minor version.

**Comment by jakepusateri** [comment]
I'm also getting these warnings, and would like to see them fixed. Most modules ship the pre-built files in a `dist` folder for those that need them and keep the CommonJS files as the default at the root.

**Comment by M-Porter** [comment]
Agreed with jake. Putting pre-built "source" files in a src directory is a red herring.

**Comment by Otouto** [comment]
+1 to resolve this issue

this is _very_ annoying and I didn`t found the workaround

**Comment by mrak** [comment]
For those of you seeing this warning with our pre-built files and wish to silence it, you can try using the [`module.noParse`](http://webpack.github.io/docs/configuration.html#module-noparse) configuration option to skip parsing `braintree-web` files. [This is suggested by the webpack team](https://github.com/webpack/webpack/issues/319#issuecomment-46897550).

```javascript
// webpack.config.js
 module.exports = {
   module: {
     noParse: [/braintree-web/]
   }
 };
```

The way this library is currently written prevents the source files form being used directly due to build-time transformations.

Ideally, we would have no build-time requirements of our source code.

**Comment by M-Porter** [comment]
This absolutely is not the suggested approach. If you would actually read what `module.noParse` is, you would see this:
> The files are expected to have no call to require, define or similar. They are allowed to use exports and module.exports.

Hate to break it to you, but your modules do call require. `package.json` points to `src/index.js` as the main entry point. [That file has `require` calls.](https://github.com/braintree/braintree-web/blob/master/src/index.js) So you're fix is not a fix at all and this should be reopened.

For anyone else coming across this on google search (like I did), you can suppress the warnings by disabling warnings, but that can be bad and you may miss more useful warnings.

To disable warnings i did something like this in my build script:
```javascript
const compiler = webpack(getConfig(buildOptions));
const compilerCallback = (err, stats) => {
    gutil.log(stats.toString({ chunks: false, warnings: false, colors: true }));
    if (!options.watch) {
        callback('complete');
    }
};
compiler.run(compilerCallback);
```

Again, I would highly suggest against suppressing warnings because you may miss legitimate warnings.

**Comment by mrak** [comment]
Thanks for the detailed response, @M-Porter. You are absolutely correct; we overlooked our `index.js` and `debug.js` files which still had references to `require`.

We have just released version `3.3.0` where these two files are correctly bundled without references to `require`.

**Comment by M-Porter** [comment]
Awesome thanks @mrak! 💃 👯

**Comment by leanne-tite-jg** [comment]
Thanks @mrak, warnings gone!

**Comment by crookedneighbor** [comment]
We've released 3.8.0, which includes the src files on NPM which means you no longer need the workaround to suppress warnings. If you had the noParse for braintree-web in your webpack config, you will need to remove that before using 3.8.0
